app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseService.kt

```diff
@@ -309,6 +311,22 @@ class FirestoreDatabaseService internal constructor(
         return documentRef.get().await().toObject(UserSchema::class.java)!!.toPublic()
     }
 
+    //TODO Optimize transaction to db
+    override suspend fun setAdminAccess(firebaseUser: FirebaseUser ,adminAccess : Boolean): User {
+        val uid = firebaseUser.uid
+        val type = User.Type.FIREBASE
+        val documentRef = users.document("$uid@$type")
+        val data = UserSchema().apply {
+            isAdmin = adminAccess
+        }
+        documentRef.set(data).await()
+        return documentRef.get().await().toObject(UserSchema::class.java)!!.toPublic()
+    }
+
+    override suspend fun checkIsAdmin(fireStoreUser: User): Boolean {
+        return fireStoreUser!!.isAdmin
```

Comment:
I don't think the non-null assertion is necessary here?

Reply:
it is not , just by reflex , I ll remove it